server/kit/kitserver.go

```diff
@@ -110,6 +121,34 @@ func NewServer(svc Service) *Server {
 }
 
 func (s *Server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	defer func() {
+		if x := recover(); x != nil {
+			var err error
+			if e, ok := x.(error); ok {
+				err = e
+			}
+
+			s.logger.Log("error", err, "message", "the server encountered a panic")
+
+			w.WriteHeader(http.StatusInternalServerError)
+			_, err = w.Write([]byte(http.StatusText(http.StatusInternalServerError)))
+			if err != nil {
+				s.logger.Log("error", err, "message", "unable to respond post-panic")
+			}
+
+			// if we have an error client, send out a report
+			if s.errs == nil {
+				return
+			}
+
+			s.errs.Report(errorreporting.Entry{
+				Req:   r,
+				Error: err,
```

Comment:
which error are you reporting? from write to `ResponseWriter`?

Reply:
good catch! I'll make a new var for the write error so it doesnt overwrite the error that caused the panic